test/serve_seeds_api.ts

```diff
@@ -0,0 +1,276 @@
+      const body = {}
+      const badWallet = Web3.eth.accounts.create('unpermittedWallet');
+      const signature = badWallet.sign(JSON.stringify(body)).signature
+
+      supertest(app).post('/').send(body)
+        .set('x-signature', signature)
+        .expect(403)
+        .end((err,res) => { if (err) throw err });
+
+    })
+  })
+
+  describe('authenticated', () => {
+    const endpoint = '/v1/seeds'
+
+    describe('an unsupported seed entity', () => {
+      it('returns an error', () => {
+        const body = { entity: 'crypto-dollars', data: { gimme: 'some' } };
+        const signature = wallet.sign(JSON.stringify(body)).signature
+
+        supertest(app).post(endpoint).send(body)
+          .set('x-signature', signature)
+          .expect(422, { error: 'unknown seed entity' })
+          .end((err,res) => { if (err) throw err });
+
+      })
+    })
+
+    describe('platforms', () => {
+      describe('with the incorrect shape', () => {
+        it('returns an error', () => {
+          const body = { entity: 'platforms', data: { blam: 'yam' } };
+          const signature = wallet.sign(JSON.stringify(body)).signature
+
+          supertest(app).post(endpoint).send(body)
+            .set('x-signature', signature)
+            .expect(422, { error: 'platforms entity is missing required fields' })
+            .end((err,res) => { if (err) throw err });
+        })
+      })
+
+      describe('with an unknown type', () => {
+        it('returns an error', () => {
+          const body = { entity: 'platforms', data: { id: 'potato', name: 'potato', type: 'yum' } }
+          const signature = wallet.sign(JSON.stringify(body)).signature
+
+          supertest(app).post(endpoint).send(body)
+            .set('x-signature', signature)
+            .expect(422, { error: 'not a valid platforms type' })
+            .end((err,res) => { if (err) throw err });
+        })
+      })
+
+      describe('with unsupported fields', () => {
+        it('returns an error', () => {
+          const body = { entity: 'platforms', data: { id: 'potato', name: 'potato', type: MusicPlatformType.sound, hackyou: 'boo' } }
+          const signature = wallet.sign(JSON.stringify(body)).signature
+
+          supertest(app).post(endpoint).send(body)
+            .set('x-signature', signature)
+            .expect(422, { error: 'platforms entity has unsupported fields' })
+            .end((err,res) => { if (err) throw err });
+        })
+      })
+
+      describe('with a valid payload', () => {
+        it('returns a 200', () => {
+          const body = { entity: 'platforms', data: { id: 'jamboni', name: 'Jamboni Jams', type: MusicPlatformType.sound } }
+          const signature = wallet.sign(JSON.stringify(body)).signature;
+
+          supertest(app).post(endpoint).send(body)
+            .set('x-signature', signature)
+            .expect(200)
+            .end((err,res) => { if (err) throw err });
+        })
+        it('persists the seed');
```

Comment:
Related to the test db setup, we might also add an assertion to make sure that the DB persists the expected seed. But then we need to setup & flush the DB after each test. 

Alternatively, I seem to remember there are some in-memory postgres emulations for testing environments? I'll wait for some advice before doing this, so just a pending test for now.

Reply:
yeah i think let's not worry about this for now?

Reply:
We can leave for now and make another pass on it if/when we do the in-memory postgres experiment